### DISCUSSION TITLE
Security concern: Vertex AI in Firebase requires no user authentication?

### DISCUSSION THREAD

**Discussion by marcoancona** [firebase/firebase-js-sdk#8558] [General]
The premise of Vertex AI in Firebase is great: being able to call AI APIs from the client side without the hassle of implementing a custom authentication layer.

According to our early tests with Vertex AI in Firebase, however, the API calls to Google services expose the Firebase API Key (which is stored client-side, [by design](https://firebase.google.com/docs/projects/api-keys)). And the API Key is everything you need to call the Gemini APIs 😳.
It seems this makes it super easy to steal the API keys of another product and use them to get access to the expensive Gemini APIs.

The documentation recommends setting up App Check, but this is 1) not mandatory, 2) not easy to setup, and 3) not practical for some environments (see Electron apps).  

I would have expected a way to limit access to the APIs to _authenticated_ users, with per-user quotas and some form of security rules similar to Firestore. But as far as I can see, no such concept exists for Vertex AI. 

Unless I missed something, it seems that these defaults and the lack of better tools to control access will push many Firebase users to leak access to their AI APIs to the world, intentionally or unintentionally.

**Comment by dpereira** [comment]
this comment feels prescient, now, given the amount of compromised keys surfacing. Wonder why no replies.
